tests/image/test_ssim.py

```diff
@@ -52,16 +60,19 @@ def _sk_ssim(preds, target, data_range, multichannel, kernel_size):
         sk_preds = sk_preds[:, :, :, 0]
         sk_target = sk_target[:, :, :, 0]
 
-    return structural_similarity(
-        sk_target,
-        sk_preds,
-        data_range=data_range,
-        multichannel=multichannel,
-        gaussian_weights=True,
-        win_size=kernel_size,
-        sigma=1.5,
-        use_sample_covariance=False,
-    )
+    results = torch.zeros(sk_preds.shape[0])
+    for i in range(sk_preds.shape[0]):
+        results[i] = structural_similarity(
+            sk_target[i],
+            sk_preds[i],
+            data_range=data_range,
+            multichannel=multichannel,
+            gaussian_weights=True,
+            win_size=kernel_size,
+            sigma=1.5,
+            use_sample_covariance=False,
+        )
+    return results
```

Comment:
I'm not sure if this implementation doesn't cause some problems. Please, refer to the failing tests @weningerleon